usefulUtilities/hiFiCalendar/clientScripts/meetingRoomSetup_app.js

```diff
@@ -0,0 +1,161 @@
+    var TOKEN_SERVER_ID = CONFIG.TOKEN_SERVER_ID;
+    var roomConfig = false;
+    var token;
+    var tokenLifetime;
+    var tokenStart;
+    var expireTime;
+    var refreshToken;
+    var timezone;
+    var timezoneOffset = 7;
+    var roomConfigInfo = [];
+
+    // Handle messages from the tablet
+    function onWebMessage(data) {
+        switch (data.type) {
+            case "EVENT_BRIDGE_OPEN_MESSAGE":
+                ui.sendToHtml({
+                    type: "AVAILABLE ROOMS",
+                    roomConfig: roomConfigInfo
+                });
+                break;          
+            case "GET AVAILABLE ROOMS":
+                ui.sendToHtml({
+                    type: "AVAILABLE ROOMS",
+                    roomConfig: roomConfigInfo
+                });
+                break;
+            case "SETUP COMPLETE":
+                token = data.access_token;
+                refreshToken = data.refresh_token;
+                tokenLifetime = data.expireTime;
+                tokenStart = data.validSince;
+                expireTime = tokenStart + tokenLifetime * MS_TO_SEC;
+                timezone = data.timeZoneName;
+                // Gets the correct timezone offset for events current / in the future, which is why the -1 is needed.
+                timezoneOffset = (new Date().getTimezoneOffset() / MIN_PER_HR) * -1; 
+                roomConfig = data.connectionData;
+                Settings.setValue("roomConfigured", true);
+                Entities.callEntityServerMethod(TOKEN_SERVER_ID, "initializeRooms", [
+                    token, 
+                    refreshToken, 
+                    expireTime, 
+                    timezone,
+                    timezoneOffset, 
+                    JSON.stringify(roomConfig)
+                ]);
+                break;
+        }
+    }
+
+
+    // This function is the listener for messages on the message mixer related to this app.
+    var messageHandler = function(channel, message, senderUUID, localOnly) {
+        if (channel !== CHANNEL) {
+            return;
+        } else {
+            try {
+                message = JSON.parse(message);
+            } catch (e) {
+                console.log(e, "Could not parse message");
+                return;
+            }
+
+            if (message.TOKEN_SERVER_ID) {
+                TOKEN_SERVER_ID = message.TOKEN_SERVER_ID;
+            }
+
+            if (message.type === "TOKEN EXPIRED") {
+                Window.alert("The authorization token for your Google Calendar could not be refreshed.\n" + 
+                "Please open your calendar app and reauthorize to continue displaying calendar schedules.");
+                roomConfig = message.roomConfig;
+                roomConfigInfo = message.roomConfigInfo;
+            } else if (message.type === "STATUS UPDATE") {
+                if (message.tokenStatus) {
+                    Window.announcement("Access tokens are valid. No action required");
```

Comment:
Is this necessary?

Reply:
Not really.  Hasn't particularly helped me.  Going to take it out.